mlflow/tracking/client.py

```diff
@@ -2681,14 +2682,17 @@ def get_model_version_stages(
         """
         return ALL_STAGES
 
-    def set_model_version_tag(self, name: str, version: str, key: str, value: Any) -> None:
+    def set_model_version_tag(
+        self, name: str, version: str = None, key: str = None, value: Any = None, stage: str = None
+    ) -> None:
         """
         Set a tag for the model version.
 
         :param name: Registered model name.
         :param version: Registered model version.
         :param key: Tag key to log.
         :param value: Tag value to log.
```

Comment:
@subramaniam02 Can we update the docstring to indicate that the tag key and value are required? This is less clear with the new backwards-compatible method signature, so supplementing the docstring would help. Thanks for taking care to make sure this is backwards-compatible :)

Reply:
Can we also add validation to ensure that, if `key` is `None` or `value` is `None`, we throw a clear exception with code `INVALID_PARAMETER_VALUE`?

Reply:
@dbczumar  agreed. Unfortunately I will be offline for next couple of days. Will take care of it when I am back online. Sorry about the delay.

Reply:
@subramaniam02 Thank you for adding validation logic for `stage` and `version`; would you be able to address the above comments about `key` and `value`?

Reply:
I forgot about that comment. Will do.